### PR TITLE
Move Azure map initialization to a seperate component

### PR DESCRIPTION
**Pull request by spryor**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
The current implementation of azure maps is giving console errors related to calling azure map apis before the map is completely initialized. This PR wraps the azure map in it's own component that will provide the map reference only after the initialization is complete. This prevents making calls to the map api before it is ready, thus removing the console errors.

### CODE REVIEW

File: src/components/pages/dashboard/panels/map/azureMap.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      if (isFunc(this.props.onMapReady)) {
+        this.props.onMapReady(this.map);
+      }
+    });
+  }
+
+  render() {
+    return <div id="map"></div>;
+  }
+
```
Comment (mechaffin) [inline src/components/pages/dashboard/panels/map/azureMap.js]: extra line

File: src/components/pages/dashboard/panels/map/mapPanel.js [hunk 2]
```diff
@@ -33,62 +34,33 @@ const deviceToMapPin = ({ id, properties, type }) =>
...
-  initializeMap(azureMapsKey) {
-    const center = new AzureMaps.data.Position(...Config.mapCenterPosition);
-    this.map = new AzureMaps.Map('map', {
-      'subscription-key': azureMapsKey,
-      center,
-      zoom: 11
-    });
-
+  onMapReady = (map) => {
+    this.map = map;
```
Comment (mechaffin) [inline src/components/pages/dashboard/panels/map/mapPanel.js]: Set this.map at the end of this function, just before the call to calculatePins.  That way there's no chance that map is there but the layers aren't when calculatePins is called from other places (like componentWillReceiveProps).

File: src/components/pages/dashboard/panels/map/azureMap.js (hunk 1 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/map/azureMap.js]: I added the newline intentionally.

File: src/components/pages/dashboard/panels/map/mapPanel.js (hunk 2 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/map/mapPanel.js]: Nice catch!

**Pull request merged**